lanelet2/lanelet2_core/src/LaneletMap.cpp

```diff
@@ -616,6 +686,37 @@ void LaneletMap::add(Area area) {
   }
 }
 
+void LaneletMap::remove(const RegulatoryElementPtr& regElem)
+{
+  if (!regElem) 
+  {
+    throw NullptrError("Empty regulatory element passed to remove()!");
+  }
+  if (regElem->id() == InvalId)
+  {
+    throw InvalidInputError("Regulatory element with InvalidId is passed to remove()!");
+  }
+  if (regulatoryElementLayer.exists(regElem->id()))
+  {
+    regulatoryElementLayer.remove(regElem->id());
+  }
+  // remove local copies of regem inside lanelet and areas
+  for (Lanelet llt : laneletLayer.findUsages(regElem))
+  {
+    llt.removeRegulatoryElement(regElem);
+  }
+  for (Area area : areaLayer.findUsages(regElem))
+  {
+    area.removeRegulatoryElement(regElem);
+  }
+  // remove its (possibly dangling) parameters 
+  // TODO: uncomment when each layer's remove is implemented
+  /*
+  RemoveVisitor visitor(this);
+  regElem->applyVisitor(visitor);
+  */
```

Comment:
If we were to remove the above section. Are you saying when we called remove(regulatoryElement) all the parameters of that element would also be removed or is it only the dangling ones? 
I would when removing a regulatory element we wouldn't want to remove parameters unless is was explicitly required. Since just because a regulation no longer applies does not mean that the road it applied to no longer exists.

Reply:
That is true. I didn't realize what I have here would end up removing every parameter unless there was a logic to filter which is dangling and which is not.